esp-hal/src/dma/mod.rs

```diff
@@ -893,6 +913,37 @@ impl DescriptorChain {
         circular: bool,
         data: *mut u8,
         len: usize,
+    ) -> Result<(), DmaError> {
+        self.fill(circular, data, len, |desc, _| {
+            desc.reset_for_rx();
+            // Descriptor::size has been set up by `link_with_buffer_impl`
```

Comment:
When is this called?

Is this clearer?
```suggestion
            // Descriptor::size has been set up in `fill`
```